src/logic/map_objects/tribes/ship.cc

```diff
@@ -2352,6 +2352,7 @@ void Ship::draw_healthbar(const EditorGameBase& egbase,
 	constexpr unsigned kBonusIconSize = 6;
 	constexpr unsigned kMaxRows = 16;
 	const unsigned bonus = get_sea_attack_soldier_bonus(egbase);
+	assert(bonus < 10000);  // Sanity check
```

Comment:
With the standard tribes, the bonus shouldn't normally be more than 330 (30*11), so I suppose 1000 should be enough (may go up to 63 columns, 378 pixels wide), 2000 abundant (125 columns, 750 pixels).